gravitee-apim-rest-api/gravitee-apim-rest-api-management-v4/gravitee-apim-rest-api-management-v4-rest/src/test/java/io/gravitee/rest/api/management/v4/rest/resource/api/ApiPlansResourceTest.java

```diff
@@ -208,21 +241,51 @@ public void shouldDeleteApiPlan() {
         verify(planService, times(1)).delete(eq(GraviteeContext.getExecutionContext()), eq(PLAN));
     }
 
-    // TODO: Fix to not care about older versions :)
+    @Test
+    public void shouldNotDeletePlanWithInsufficientRights() {
+        doReturn(false)
+            .when(permissionService)
+            .hasPermission(eq(GraviteeContext.getExecutionContext()), eq(RolePermission.API_PLAN), eq(API), any());
+
+        final Response response = rootTarget().path(API).path("plans").path(PLAN).request().delete();
+        assertEquals(FORBIDDEN_403, response.getStatus());
+    }
+
+    @Test
+    public void shouldNotUpdatePlanWithInsufficientRights() {
+        doReturn(false)
+            .when(permissionService)
+            .hasPermission(eq(GraviteeContext.getExecutionContext()), eq(RolePermission.API_PLAN), eq(API), any());
+
+        final Response response = rootTarget().path(API).path("plans").path(PLAN).request().put(Entity.json(""));
+        assertEquals(FORBIDDEN_403, response.getStatus());
+    }
+
+    @Test
+    public void shouldNotPublishPlanWithInsufficientRights() {
+        doReturn(false)
+            .when(permissionService)
+            .hasPermission(eq(GraviteeContext.getExecutionContext()), eq(RolePermission.API_PLAN), eq(API), any());
+
+        final Response response = rootTarget().path(API).path("plans").path(PLAN).path("_publish").request().post(Entity.json(""));
+        assertEquals(FORBIDDEN_403, response.getStatus());
+    }
+
+    @Test
+    public void shouldNotDeprecatePlanWithInsufficientRights() {
+        doReturn(false)
+            .when(permissionService)
+            .hasPermission(eq(GraviteeContext.getExecutionContext()), eq(RolePermission.API_PLAN), eq(API), any());
+
+        final Response response = rootTarget().path(API).path("plans").path(PLAN).path("_deprecate").request().post(Entity.json(""));
+        assertEquals(FORBIDDEN_403, response.getStatus());
+    }
+
     private ApiEntity getApi(DefinitionVersion version) {
```

Comment:
As we only manage V4 definition, what do you think of hardcoding V4 in the method and removing the parameter from the signature?

Reply:
Yes that works!